app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,8 @@
+class PagesController < ApplicationController
+  def index
+    @works = Work.order(votes_count: :desc)
+    @books = @works.where(category: 'book')
+    @albums = @works.where(category: 'album')
```

Comment:
While I still think `top_ten` or `spotlight` would make a good model method, using the counter cache makes this controller code pretty clean - you're definitely still offloading a lot of the work to the model.

It sounds like you've identified this, but the interesting piece you're missing by not having this logic isolated is testing. For example, for the spotlight, I would wonder:

- What happens if there are no works?
- What happens if there are works but no votes?
- What happens if two works have the same number of votes?

Similarly for the top ten, I would ask:

- What if there are no works of that category?
- What if there are less than 10 works?
- What if there's a tie for last place, e.g. works 9, 10 and 11 all have 0 votes?